packages/broker/test/integration/plugins/operator/smartContractUtils.ts

```diff
@@ -15,26 +15,38 @@ export function getProvider(): Provider {
     return new JsonRpcProvider(CONFIG.rpcEndpoints[0].url)
 }
 
-export function getTokenContract(): IERC677 {
-    return new Contract(CONFIG.contracts.LINK, tokenABI) as unknown as IERC677
+export function getTokenContract(): TestToken {
+    return new Contract(CONFIG.contracts.LINK, tokenABI) as unknown as TestToken
 }
 
-export async function generateWalletWithGasAndTokens(provider: Provider): Promise<Wallet> {
+export async function generateWalletWithGasAndTokens(provider: Provider, config?: Chain, adminKey?: string): Promise<Wallet> {
     const newWallet = new Wallet(fastPrivateKey())
-    const adminWallet = new Wallet(ADMIN_WALLET_PK).connect(provider)
-    const token = getTokenContract().connect(adminWallet)
-    for (let i = 0; i < 5; i++) {
-        try {
-            // eslint-disable-next-line no-console
-            // console.log("trying with nonce " + await adminWallet.getTransactionCount() + " time " + new Date().getTime() / 1000)
-            await (await token.transfer(newWallet.address, parseEther("1000"), {
-                nonce: await adminWallet.getTransactionCount()
-            })).wait()
-            break
-        } catch (e) {
-            await new Promise((resolve) => setTimeout(resolve, 3000))
-            // eslint-disable-next-line no-console
-            // console.log("sending link failed, retrying")
+    const adminWallet = new Wallet(adminKey ?? ADMIN_WALLET_PK).connect(provider)
+
+    if (config && !config.contracts.LINK) {
+        const token = new Contract(config.contracts.DATA, tokenABI, adminWallet) as unknown as TestToken 
+        // eslint-disable-next-line no-console
+        console.log("trying data with nonce " + await adminWallet.getTransactionCount() + " time " + new Date().getTime() / 1000)
+        await (await token.mint(newWallet.address, parseEther("1000000"), {
+            nonce: await adminWallet.getTransactionCount()
+        })).wait()
+        // eslint-disable-next-line no-console
+        console.log("sent Data to " + newWallet.address)
```

Comment:
```suggestion
        //console.log("sent Data to " + newWallet.address)
```
or make trace level log statement